### PR TITLE
[Havoc Demon Hunter] Felblade timeline fix and dps added to stat box

### PR DESCRIPTION
**Pull request by philipsabri**
- DPS added to the Statistic Box
- Timeline now showing felbalde casted correctly

Before:
image image

Now:
image image

### CODE REVIEW

File: src/parser/core/CASTS_THAT_ARENT_CASTS.ts [hunk 1]
```diff
@@ -58,6 +58,10 @@ const spells: number[] = [
   SPELLS.COMBAT_MEDITATION_TRIGGER.id, //The Spell ID when Combat Meditation from Kyrian Soulbind triggers, so that there is now a soul ready to pick up
   SPELLS.COMBAT_MEDITATION_PICK_UP.id, //The Spell ID that is "cast" when an orb is picked up
   //endregion
+
+  //region Demon Hunter
+  SPELLS.FELBLADE_PAIN_GENERATION.id, //The spell that generates fury when casting 
+  //endregion
 ];
```
Comment (Pewtro) [inline src/parser/core/CASTS_THAT_ARENT_CASTS.ts:65]: Spell should be renamed to match it generating fury and not pain

File: src/parser/demonhunter/havoc/modules/talents/Felblade.js [hunk 2]
```diff
@@ -41,13 +42,18 @@ class Felblade extends Analyzer {
...
   }
 
   onEnergizeEvent(event) {
     this.furyGain += event.resourceChange;
     this.furyWaste += event.waste;
   }
 
+  onDamageEvent(event) {
+    this.damage += event.amount;
+  }
```
Comment (Pewtro) [inline src/parser/demonhunter/havoc/modules/talents/Felblade.js:56]: damage events should include absorbed damage. ``this.damage += event.amount + (event.absorbed || 0);``

File: src/parser/demonhunter/havoc/modules/talents/Felblade.js [hunk 3]
```diff
@@ -69,13 +75,15 @@ class Felblade extends Analyzer {
...
+            {formatThousands(this.damage)} Total damage
           </>
         )}
       >
         <BoringSpellValueText spell={SPELLS.FELBLADE_TALENT}>
           <>
-            {this.furyPerMin} <small>Fury per min </small>
+            {this.furyPerMin} <small>Fury per min </small><br />
+            {this.owner.formatItemDamageDone(this.damage)}
           </>
```
Comment (Pewtro) [inline src/parser/demonhunter/havoc/modules/talents/Felblade.js:88]: You can also use ``<ItemDamageDone amount={this.damage}/>`` for a nice added sword to the metric to match rest of the codebase

File: src/parser/core/CASTS_THAT_ARENT_CASTS.ts (hunk 1 above)
Comment (philipsabri) [reply src/parser/core/CASTS_THAT_ARENT_CASTS.ts:65]: True. I noticed its both fury gen + damage. Would FELBALDE_DAMAGE be accepted in such case?

File: src/parser/demonhunter/havoc/modules/talents/Felblade.js (hunk 3 above)
Comment (philipsabri) [reply src/parser/demonhunter/havoc/modules/talents/Felblade.js:88]: Oh, looks better!

File: src/parser/demonhunter/havoc/modules/talents/Felblade.js (hunk 2 above)
Comment (philipsabri) [reply src/parser/demonhunter/havoc/modules/talents/Felblade.js:56]: Fixed

File: src/parser/core/CASTS_THAT_ARENT_CASTS.ts (hunk 1 above)
Comment (Pewtro) [reply src/parser/core/CASTS_THAT_ARENT_CASTS.ts:65]: Sure - with correct felblade spelling 😉

Comment (philipsabri) [reply src/parser/core/CASTS_THAT_ARENT_CASTS.ts:65]: Haha its correct in the code 💯 (latest commit)

**Pull request merged**